scripts/SANS/sans/gui_logic/presenter/run_tab_presenter.py

```diff
@@ -38,7 +39,7 @@
 from sans.gui_logic.models.settings_adjustment_model import SettingsAdjustmentModel
 from sans.gui_logic.models.state_gui_model import StateGuiModel
 from sans.gui_logic.models.table_model import TableModel
-from sans.gui_logic.presenter.Observers.run_tab_observers import SaveOptionsObserver, RunTabObservers
+from sans.gui_logic.presenter.Observers.run_tab_observers import RunTabObservers
```

Comment:
Good questions:

> [Replacing] SaveOptionsObserver with generic observers

This is because unlike C++, Python uses duck typing. So having a class per observer provides no real advantages. They all act and behave like generic observers in the end. Rather than having a file of stub classes I decided to just use the generic observer instead

> why continue to use RunTabObservers

`RunTabObservers` is a dataclass which simply holds all the different observer types `run_tab_presenter` can require. Ideally we would just put them into the presenter, but as our `run_tab_presenter` is effectively 10+ widgets bundled into one "god-ui" I've been trying to gradually chunk things out rather than add to it. For this reason it makes sense (to me at least) to group them together into a class since I can foresee long-term us moving more into this observer pattern to match project GUI standards.
I'm happy to shuffle the members somewhere else though, or any other ideas?

Reply:
This was more a query to understand what was going on rather than a prelude to suggested changes! It does sound like a sensible change under the circumstances and I have no idea where else these could be shuffled to or any other ideas. Need to look up duck typing....

Reply:
Duck typing is a Python concept, in a nutshell:
"If it looks like a duck, sounds, quacks and waddles like one it's probably a duck"

Well our SaveOptionsObserver looked, acted and had the API of a GenericSaveObserver so it's easier to reason with it just being a GenericSaveObserver